draco/core/containers.py

```diff
@@ -265,6 +265,32 @@ def add_dataset(self, name):
 
         return dset
 
+    def chunkify(self):
+        """Ensure datasets that have chunk/compression specs are chunked.
+
+        For every dataset, check if chunks and compression are set, and
+        if not set them to dataset_spec values.
+        """
+        for dset in self.dataset_spec:
+            if "chunks" in self.dataset_spec[dset] and self[dset].chunks is None:
+                # ensure chunks aren't larger than dataset shape
+                chunks = ()
+                for i, l in enumerate(self[dset].shape):
+                    chunks += (min(self.dataset_spec[dset]["chunks"][i], l),)
```

Comment:
What is this doing?
Why would this be below 1 or why would you choose chunk size 1?

Reply:
did you misread `l` as `1`?

Reply:
Thanks